### PR TITLE
WIP: Add support for Gitea

### PR DESCRIPTION
**Pull request by Alexsaphir**
Added support for Gitea provider following the work of SimonVHB.

### CODE REVIEW

**Comment by mortbauer** [comment]
Does this work with the current gitea version or do you use some special branch?

**Comment by jimsheldon** [comment]
@Alexsaphir this is looking good! The failure is unrelated, we need to add docker hub authentication for our pull requests.

Can you look at how we have added tests for bitbucket and github? Can you add a test for your `GetGiteaFilesChanged` function? Thanks

File: providers/gitea_server.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	var changes []*scm.Change
+	var result *scm.Response
+
+	if build.Before == "" || build.Before == scm.EmptyCommit {
+		changes, result, err = client.Git.ListChanges(newctx, repo.Slug, build.After, scm.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		changes, result, err = client.Git.CompareChanges(newctx, repo.Slug, build.Before, build.After, scm.ListOptions{})
```
Comment (colinhoglund) [inline providers/gitea_server.go:46]: FYI, neither of these calls are implemented in go-scm.

https://github.com/drone/go-scm/blob/be60c693c948b2746bd1c68684dcc51478757699/scm/driver/gitea/git.go#L76-L82

**Comment by jimsheldon** [comment]
I am going to close this for now, if you would like to re-submit, we have recently added a [contributing](https://github.com/meltwater/drone-convert-pathschanged/blob/master/CONTRIBUTING.md) doc which we would like to follow.

We have been slowly adding support for other SCM solutions, we would certainly be open to adding Gitea support.

**Pull request closed without merging**